From 8ca77ffe4ea5af8590ed407eb23a4126ca380ae8 Mon Sep 17 00:00:00 2001
From: Nathan Morrisson <nmorrisson@phytec.com>
Date: Fri, 6 Mar 2026 15:14:08 -0800
Subject: [PATCH] WIP bugfix

Signed-off-by: Nathan Morrisson <nmorrisson@phytec.com>
---
 drivers/input/joystick/phyhandheld.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/input/joystick/phyhandheld.c b/drivers/input/joystick/phyhandheld.c
index a790b07e4697..fdfd4902f648 100644
--- a/drivers/input/joystick/phyhandheld.c
+++ b/drivers/input/joystick/phyhandheld.c
@@ -215,7 +215,7 @@ static int phyhandheld_read_data5(struct i2c_client *client, struct phyhandheld_
 	}
 
 	if (data->j2_ad2 >= 3429) {
-		data->j2_ad2 = 0x3429;
+		data->j2_ad2 = 3429;
 	} else if (data->j2_ad2 <= 700) {
 		data->j2_ad2 = 701;
 	}
@@ -223,10 +223,10 @@ static int phyhandheld_read_data5(struct i2c_client *client, struct phyhandheld_
 	dev_dbg(&client->dev, "2 ----- phyhandheld_register_read_data: %d %d %d %d\n",
 				data->j1_ad1, data->j1_ad2, data->j2_ad1, data->j2_ad2);
 
-	data->j1_ad1 = (be16_to_cpu(read_buf[1]) * 3) / 2 - 1050;
-	data->j1_ad2 = (be16_to_cpu(read_buf[2]) * 3) / 2 - 1050;
-	data->j2_ad1 = (be16_to_cpu(read_buf[3]) * 3) / 2 - 1050;
-	data->j2_ad2 = (be16_to_cpu(read_buf[4]) * 3) / 2 - 1050;
+	data->j1_ad1 = (data->j1_ad1 * 3) / 2 - 1050;
+	data->j1_ad2 = (data->j1_ad2 * 3) / 2 - 1050;
+	data->j2_ad1 = (data->j2_ad1 * 3) / 2 - 1050;
+	data->j2_ad2 = (data->j2_ad2 * 3) / 2 - 1050;
 
 	if (data->j1_ad1 > 1500 && data->j1_ad1 < 2500 && data->j1_ad2 > 1500 && data->j1_ad2 < 2500){
 		data->j1_ad1 = 2000;
-- 
2.43.0

