From d328ba1d03bd5af79e2c068c5b5110a8a0d93dbd Mon Sep 17 00:00:00 2001
From: Nathan Morrisson <nmorrisson@phytec.com>
Date: Thu, 5 Mar 2026 16:12:23 -0800
Subject: [PATCH] WIP joystick scaling

Signed-off-by: Nathan Morrisson <nmorrisson@phytec.com>
---
 drivers/input/joystick/phyhandheld.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/input/joystick/phyhandheld.c b/drivers/input/joystick/phyhandheld.c
index c3c2bf4648eb..150a1ca47acf 100644
--- a/drivers/input/joystick/phyhandheld.c
+++ b/drivers/input/joystick/phyhandheld.c
@@ -191,6 +191,11 @@ static int phyhandheld_read_data5(struct i2c_client *client, struct phyhandheld_
 	data->j2_ad1 = be16_to_cpu(read_buf[3]);
 	data->j2_ad2 = be16_to_cpu(read_buf[4]);
 
+	data->j1_ad1 = (be16_to_cpu(read_buf[1]) * 3) / 2;
+	data->j1_ad2 = (be16_to_cpu(read_buf[2]) * 3) / 2;
+	data->j2_ad1 = (be16_to_cpu(read_buf[3]) * 3) / 2;
+	data->j2_ad2 = (be16_to_cpu(read_buf[4]) * 3) / 2;
+
 	return 0;
 }
 
-- 
2.43.0

